File: backend/app/src/models/buildings/IronMine.py
from .Mine import Mine


class IronMine(Mine):
    def __init__(self, workers: list = []) -> None:
        super().__init__(name="Iron Mine", worker_slots=2, workers=workers)
        
    
    def convert_worker_to_miner(self, worker, location) -> bool:
        if self.convert_worker_to_craftsman("Iron Mine", worker, location):
            return True

        return False







if __name__ == "__main__":
    pass

File: backend_refactoring/CSharpApi/Models/BluePrints/BuildingTypes/IronMine.cs


using CSharpApi.Models.BluePrints.Beings;

namespace CSharpApi.Models.BluePrints.BuildingTypes
{
    public class IronMine : Mine, IBuildingList
    {
        public string Name { get; set; } = "Iron Mine";

        private List<Worker> _workers;

        public Dictionary<string, Dictionary<string, float>> Costs { get; set; } = new() {
            { "costs", new()
                {
                    { "gold", 200 },
                    { "food", 100 },
                    { "wood", 200 },
                    { "iron", 50 }
                }
            }
        };

        public IronMine(List<Worker> workers = null)
           : base("Iron Mine", true, 2, workers ?? [])
        {
            _workers ??= workers ?? [];
        }

        public bool ConvertWorkerToMiner(int workerId, object location)
        {
            return (bool)ConvertWorkerToCraftsman(Name, workerId, location)["success"];
        }
    }
}
